openstack/networking/v2/extensions/attributestags/results.go

```diff
@@ -38,3 +38,13 @@ type DeleteResult struct {
 type AddResult struct {
 	gophercloud.ErrResult
 }
+
+// Extract interprets ConfirmResult to return bool
+func (r ConfirmResult) Extract() (bool, error) {
+	return r.Err == nil, r.Err
```

Comment:
I apologize, I might not have looked closely enough at this last time.

The logic to determine if the tag exists makes sense to me, but there's a possible issue where a true error is ignored because the user focused only on the `bool` portion. For example, what if a 401 or 500 was returned?

How about something like this:

```go
func (r ConfirmResult) Extract() (bool, error) {
  exists := r.Err == nil

  if r.Err != nil {
    if _, ok := r.Err.(gophercloud.ErrDefault404); ok {
      r.Err = nil
    } 
  } 

  return exists, r.Err
}
```

So we're considering 404 errors to be acceptable and setting them to `nil`. This way a user can now do:

```go
exists, err := attributestags.Confirm(client, "networks", network.ID, "a").Extract() 
if err != nil {
  panic(err)
}

if !exists {
  panic("Tag does not exist")
}
```

Thoughts?

Reply:
Yes good point, I was mostly thinking of the 404 case but you're right we do need to handle other errors - I'll update it as you describe, thanks!